### PR TITLE
Upgrade app-draft-cache Terraform from 0.11.15 to 0.12.30

### PR DESCRIPTION
**Pull request by AgaDufrat**
## Context

It's not possible to plan/apply infrastructure changes in this project which uses terraform version 0.11 as it's relies on  `terraform/modules/aws/alarms/elb/main.tf`, which had its syntax updated to 0.12.
Running terraform plan resulted in 
```
Error: Error parsing /var/lib/jenkins/workspace/Deploy_Terraform_GOVUK_AWS/terraform/projects/app-draft-cache/main.tf:
At 285:37: unexpected token while parsing list: IDENT
```
https://deploy.integration.publishing.service.gov.uk/job/Deploy_Terraform_GOVUK_AWS/4965/console

## What
This updates the terraform version to 0.12.30 and applies formatting changes (with `terraform fmt` not sure if I need to reformat all the projects)

Tested by running `plan`. See [console output](https://deploy.integration.publishing.service.gov.uk/job/Deploy_Terraform_GOVUK_AWS/5041/console)

### CODE REVIEW

**Review by ChrisBAshton — COMMENTED** [review]
Thanks @AgaDufrat - changes look good, some minor nitpicks below and one larger suggestion. ⭐

File: terraform/projects/app-account/main.tf [hunk 1]
```diff
@@ -55,7 +55,7 @@ variable "instance_type" {
 # Resources
 # --------------------------------------------------------------
 terraform {
-  backend          "s3"             {}
+  backend "s3" {}
```
[the remark this answers is not in the document]
Comment (AgaDufrat) [reply terraform/projects/app-account/main.tf:58]: Yeah, it was the `terraform fmt` failing in the CI. I included this info in the commit message as suggested.

Screenshot 2023-02-13 at 10 00 16

File: terraform/modules/aws/node_group/main.tf [hunk 2]
```diff
@@ -370,7 +370,7 @@ resource "aws_autoscaling_attachment" "node_autoscaling_group_attachment_alb" {
 resource "aws_autoscaling_attachment" "node_autoscaling_group_attachment_classic" {
   count                  = "${var.instance_elb_ids_length}"
   autoscaling_group_name = "${aws_autoscaling_group.node_autoscaling_group.id}"
-  elb                    = "${element(var.instance_elb_ids, count.index)}"
+  elb                    = "${flatten(var.instance_elb_ids)[count.index]}"
```
[the remark this answers is not in the document]
Comment (AgaDufrat) [reply terraform/modules/aws/node_group/main.tf]: Good point, it does just need `flatten`.

File: terraform/projects/fastly-datagovuk/main.tf [hunk 3]
```diff
@@ -172,7 +172,7 @@ resource "fastly_service_v1" "datagovuk" {
   }
 
   s3logging {
-    format             = "%h\\t%{%Y-%m-%d %H:%M:%S}t.%{msec_frac}t\\t%m\\t%U%q\\t%>s\\t%B\\t%{tls.client.protocol}V\\t%{fastly_info.state}V\\t%{Referer}i\\t%{User-Agent}i"
+    format             = "%h\\t%%{%Y-%m-%d %H:%M:%S}t.%%{msec_frac}t\\t%m\\t%U%q\\t%>s\\t%B\\t%%{tls.client.protocol}V\\t%%{fastly_info.state}V\\t%%{Referer}i\\t%%{User-Agent}i"
```
[the remark this answers is not in the document]
Comment (AgaDufrat) [reply terraform/projects/fastly-datagovuk/main.tf]: I think we still need to do all the escaping – I included a link to the docs in the commit message. I also added a separate refactor commit (to minimise the diff).

File: terraform/projects/app-draft-cache/main.tf [hunk 4]
```diff
@@ -235,7 +236,7 @@ resource "aws_elb" "draft-cache_external_elb" {
 }
 
 resource "aws_route53_record" "draft-cache_external_service_record" {
-  count = var.create_external_elb
+  count = var.create_external_elb ? 1 : 0
```
Comment (ChrisBAshton) [inline terraform/projects/app-draft-cache/main.tf:239]: Nitpick: Interestingly I can't see any `create_external_elb = false` in govuk-aws-data, so this extra logic is probably [YAGNI](https://martinfowler.com/bliki/Yagni.html) and could be deleted.

File: terraform/projects/fastly-datagovuk/main.tf [hunk 5]
```diff
@@ -172,7 +172,21 @@ resource "fastly_service_v1" "datagovuk" {
...
+        "%m",
+        "%U%q",
+        "%>s",
+        "%B",
+        "%%{tls.client.protocol}V",
+        "${fastly_info.state}V",
+        "%\"Referer\"i",
+        "%\"User-Agent\"i",
+      ]
+    )}"
```
Comment (ChrisBAshton) [inline terraform/projects/fastly-datagovuk/main.tf:189]: Does this `plan` OK in Jenkins? Feel free to split it out into its own PR if it's risky. But if it is error-free, it should give the same `plan` output as the earlier commit that introduces the double `%`. 👍

**Review by ChrisBAshton — APPROVED** [review]
Thanks @AgaDufrat - LGTM! ✅

File: terraform/projects/app-draft-cache/main.tf (hunk 4 above)
Comment (AgaDufrat) [reply terraform/projects/app-draft-cache/main.tf:239]: yeah, it was used for some ["training jumpbox"](https://github.com/alphagov/govuk-aws-data/blob/b12b9fb4ae62c428521eb7cb013f385fb4728a5c/data/app-jumpbox/training/common.tfvars).
I didn't want to go too far in this PR given it already has a few commits. I'll remove it while I'm here 😉

File: terraform/projects/fastly-datagovuk/main.tf (hunk 5 above)
Comment (AgaDufrat) [reply terraform/projects/fastly-datagovuk/main.tf:189]: Compared the plans and they look the same

**Pull request merged**